modules/apps/27-interchain-accounts/controller/ibc_middleware.go

```diff
@@ -138,7 +140,10 @@ func (im IBCMiddleware) OnRecvPacket(
 	packet channeltypes.Packet,
 	_ sdk.AccAddress,
 ) ibcexported.Acknowledgement {
-	return channeltypes.NewErrorAcknowledgement("cannot receive packet on controller chain")
+	err := fmt.Errorf("cannot receive packet on controller chain")
```

Comment:
Should we be creating an SDK error with a code here?

Reply:
We're getting the error code here and adding it to the string. AFAIK we need to do it this way for determinism? 

https://github.com/cosmos/ibc-go/pull/1565/files#diff-7a038c27f47c6c3093e19a350a851ccbc15c5ef7036060e3c631a73579acc333R35

Reply:
in this particular case the only thing that actually matters is if the error is nil or not, but we use the sdkerrors.Wrap to create this error for consistency.

Reply:
I think we should use `icatypes.ErrInvalidChannelFlow`.

See: https://github.com/cosmos/ibc-go/blob/main/modules/apps/27-interchain-accounts/controller/ibc_middleware.go#L79